### ISSUE TITLE
feature request: to trigger backup manually

### ISSUE THREAD

**Issue by MeCias** [owncloud/updater#75]
Hi guys,

In situations where you have to update manually (from OC7 to OC 8) it would be great to trigger a backup manually as well.
At the moment - I think - there is just the automatic update for minor updates like 7.0.3 to 7.0.4.

Thanks in advance for considering

Cheers

**Comment by rakekniven** [comment]
Fully support feature request. 
Would be nice to have something like a backup task with schedule and called by cron.
Next step would be a selection on what to backup:
- config
- install
- themes
- data
- ...

**Comment by te-online** [comment]
@MeCias Well, that shouldn't be magic to add a button that does a backup only. In other words: It's easily possible but needs some lines of code to add an interface for it. I'll see what I can do to add this.

@rakekniven I'm not sure whether further optimization in terms of the *backup function* may heavily speak for the need to put them into a *new app*. To create and maintain this new app would of course produce a significant amount of extra work.

**Another thing:** To backup the *core* is what happens now in the updater app and could also be made selectable (config, install, themes etc.) in the future (in my opinion). To backup the *data folder*, I think, we need some more programming logic. Imagine users with terrabytes of data in their owncloud. Probably they don't want this data in one single zip-file (if that's even possible), besides maybe the free space is not big enough to backup all the data. A sophisticated backup-script for the *data-part* would be great, but also means a lot of planning, thinking and testing (correct me if that's wrong). 

Central point is the question how users *would like* to backup their data. Maybe they like to use rsync to transfer their whole owncloud to another server every hour (which wouldn't need a backup-app at all). But maybe they want to download a zip-file from the interface each week, because they only have 150 mb of data. Maybe they want to have another folder on their server where each hour a zip-file of every folder from their owncloud-data is created. Or maybe they want something else.

Problem is that we can only speculate about use-cases. Best way would be to evaluate the backup-question (e.g. with a survey) before starting to add something complex, I think. What do you think?

Maybe @VicDeo wants to say something about the general idea ;-)?

**Comment by te-online** [comment]
I added a button to do a manual backup at any time. This backup is the same backup that would be created on update.
Additionally, backups are now sorted by filedate to have the latest file on top.

![bildschirmfoto 2015-02-25 um 22 40 49](https://cloud.githubusercontent.com/assets/4127167/6381183/78ee7e6e-bd3f-11e4-9915-01a7b9982c3e.png)

I will create a pull request that reflects this change and the changes mentioned in https://github.com/owncloud/updater/issues/40 and see what @jancborchardt and @VicDeo think about this. :-)

**Comment by MeCias** [comment]
Hi,

I like it already. Thanks a lot te-online.

**Comment by jancborchardt** [comment]
@te-online looks great! :) Only thing is that I would put the »Delete« button to the right of the list, and make the »Sicherung« column aligned left with the rest of the text above (The Sicherungsverzeichnis and the button).

And, as mentioned in the other pull request, better separate the two changes.

**Comment by te-online** [comment]
@jancborchardt Okay, I see your point. 

I intentionally moved the trash-button to the left, because it needs a very long mouse-movement to delete an entry if you're sitting in front of a wide screen. 
On the other hand, the *1st important task* might be to download the file (»Sicherung«-column) and the *2nd important task* is to remove a file.

So what about the trash in between the first two columns? Doesn't sound good, either, does it?
Maybe we should keep it to the right then? :D
Or we could set a max-width...

I'm not sure whether one of the solutions is a *good* one:

![bildschirmfoto 2015-02-26 um 16 49 58](https://cloud.githubusercontent.com/assets/4127167/6395336/dcfe06f2-bdd7-11e4-8730-589407497ede.png)
![bildschirmfoto 2015-02-26 um 16 50 35](https://cloud.githubusercontent.com/assets/4127167/6395337/dcfe05b2-bdd7-11e4-9a6b-df5418dbc9dc.png)
![bildschirmfoto 2015-02-26 um 16 51 43](https://cloud.githubusercontent.com/assets/4127167/6395338/dd02bca6-bdd7-11e4-9696-800eb045b0b5.png)
![bildschirmfoto 2015-02-26 um 16 55 22](https://cloud.githubusercontent.com/assets/4127167/6395422/4fa7ae56-bdd8-11e4-9787-d23fa7fc8d0e.png)

**Comment by jancborchardt** [comment]
@te-online making the whole table narrower looks good! Also, with ownCloud 8 the admin area has a sidebar on the left, so it will look less strange (less whitespace on the right). :)

**Comment by te-online** [comment]
@jancborchardt okay, so you prefer number 3 or number 4? I can't decide ;-)
(Besides, personally I think, we are missing some more interface-styleguides in general).

**Comment by jancborchardt** [comment]
@te-online definitely number 4. :)

Yeah you’re right, we are missing those. However the general style already looks good by using the proper CSS. And defining a style for these special cases is what we have to do first before we can build a styleguide.

**Comment by te-online** [comment]
@jancborchardt okay, thanks for your answer :-)
Considering the two given opinions, I will implement the 4th option for now (in the *manual-update* branch).

**Comment by jancborchardt** [comment]
Thank you!

**Comment by te-online** [comment]
@jancborchardt I added the small visual changes into another branch, so that we can merge them while I still want to improve the code quality of the manual backup option.

**Comment by jancborchardt** [comment]
@te-online good stuff! :)

**Comment by enoch85** [comment]
Love this! Any update?

**Comment by te-online** [comment]
Forgot about that, sorry! Give me a moment for the try of a pull request :smile:

**Comment by te-online** [comment]
Okay, it's all there in the *manual-backup* branch. But I will need some time to adapt that to the code changes that were made in the meantime. Additionally, my approach is a bit hacky right now – I just copied the methods from the update-mechanism and removed everything I think is not necessary for the backup task.

Give me some days and will have another look at it :wink:

**Comment by enoch85** [comment]
@te-online :beers:

**Comment by te-online** [comment]
Wow, a few days became a month now. It was close to be right on top of my todo list multiple times, but I didn't have time to do it, yet. I am very sorry! I will definitely come back to this and it won't take another month. *wink*

**Comment by te-online** [comment]
Well, I did a fix to my previous approach (route disappeared) and it works now.

However, it doesn't work with any of the stable owncloud versions, I think, because updater introduced *\OCP\Util::getChannel()* which can't be found in the stable releases, when I try.

Note: the feature only backups **core** and no data or settings!
And again, I like someone to check code quality because I am not quite up-to-date with the coding standards here.

@enoch85 try it out. If you're trying with a stable version and receive 
``Call to undefined method OCP\Util::getChannel() at /Applications/MAMP/htdocs/ocdev8.1/apps/updater/lib/channel.php#42 
``
just comment out that line. Updates won't work than.

**Comment by enoch85** [comment]
@te-online Notcied. Don't have a  test VM setup now.

**Comment by VicDeo** [comment]
9.0 only
`sudo -u wwwrun php /path/to/owncloud/updater/application.php upgrade:checkpoint --create`
to your root crontab

or 
`php /path/to/owncloud/updater/application.php upgrade:checkpoint --create`
to the  crontab of the oc files owner

**Comment by VicDeo** [comment]
sudo -u wwwrun php /path/to/owncloud/updater/application.php upgrade:checkpoint --create
or this button
![btn](https://cloud.githubusercontent.com/assets/991300/13086347/dfd016ca-d4f3-11e5-9c75-62674c656f3c.png)

Closing as done for 9.0
